addons/account/models/account_payment.py

```diff
@@ -381,40 +378,42 @@ def _compute_is_internal_transfer(self):
             payment.is_internal_transfer = payment.partner_id == payment.journal_id.company_id.partner_id
 
     @api.depends('payment_type', 'journal_id')
-    def _compute_payment_method_id(self):
-        ''' Compute the 'payment_method_id' field.
+    def _compute_payment_method_line_id(self):
+        ''' Compute the 'payment_method_line_id' field.
         This field is not computed in '_compute_payment_method_fields' because it's a stored editable one.
         '''
         for pay in self:
             if pay.payment_type == 'inbound':
-                available_payment_methods = pay.journal_id.inbound_payment_method_line_ids.mapped('payment_method_id')
+                available_payment_methods = pay.journal_id.inbound_payment_method_line_ids
             else:
-                available_payment_methods = pay.journal_id.outbound_payment_method_line_ids.mapped('payment_method_id')
+                available_payment_methods = pay.journal_id.outbound_payment_method_line_ids
 
             # Select the first available one by default.
-            if pay.payment_method_id in available_payment_methods:
-                pay.payment_method_id = pay.payment_method_id
+            if pay.payment_method_line_id in available_payment_methods:
+                pay.payment_method_line_id = pay.payment_method_line_id
             elif available_payment_methods:
-                pay.payment_method_id = available_payment_methods[0]._origin
+                pay.payment_method_line_id = available_payment_methods[0]._origin
             else:
-                pay.payment_method_id = False
+                pay.payment_method_line_id = False
 
     @api.depends('payment_type',
                  'journal_id.inbound_payment_method_line_ids',
                  'journal_id.outbound_payment_method_line_ids')
-    def _compute_payment_method_fields(self):
```

Comment:
FYI, removing a method doesn't respect the stable policy. 

See: https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean